src/drag_face.rs

```diff
@@ -105,9 +105,9 @@ pub(crate) fn drag_face(
 /// Find the closest pair of points `(p1, p2)` where `p1` is on ray `r1` and
 /// `p2` is on ray `r2`. Returns `(t1, t2)` such that `p_n =
 /// r_n.get_point(t_n)`.
-fn closest_points_on_two_rays(r1: &Ray, r2: &Ray) -> Option<(f32, f32)> {
+fn closest_points_on_two_rays(r1: &Ray3d, r2: &Ray3d) -> Option<(f32, f32)> {
```

Comment:
IDK why this change is happening, but can we maybe localize the `.xyz()` call to the caller of `closest_points_on_two_rays`?

Reply:
This happens because bevy introduced a `Direction3d` type that wraps a `Vec3` for the ray direction. Calling `.xyz()` just gets the underlying `Vec3`. I just realized I can deref it to get the `Vec3` instead of using `.xyz()`, although, I would expect the compiler to optimize it the same way honestly.